MapleServer2/Tools/InventoryController.cs

```diff
@@ -38,12 +38,26 @@ public static void Add(GameSession session, Item item, bool isNew)
                     return;
                 }
             }
+
+            session.Player.Inventory.Add(item); // Adds item into internal database
+            session.Send(ItemInventoryPacket.Add(item)); // Sends packet to add item clientside
+            if (isNew)
+            {
+                session.Send(ItemInventoryPacket.MarkItemNew(item, item.Amount)); // Marks Item as New
+            }
         }
-        session.Player.Inventory.Add(item); // Adds item into internal database
-        session.Send(ItemInventoryPacket.Add(item)); // Sends packet to add item clientside
-        if (isNew)
+        else
         {
-            session.Send(ItemInventoryPacket.MarkItemNew(item, item.Amount)); // Marks Item as New
+            for (int i = 0; i < item.Amount; i++)
+            {
+                Item temp = Item.CreateNew(item);
+                session.Player.Inventory.Add(temp); // Adds item into internal database
+                session.Send(ItemInventoryPacket.Add(temp)); // Sends packet to add item clientside
+                if (isNew)
+                {
+                    session.Send(ItemInventoryPacket.MarkItemNew(temp, temp.Amount)); // Marks Item as New
```

Comment:
We probably don't need to comment these lines when the exact same lines are above with comments